src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java

```diff
@@ -169,6 +172,22 @@ public void handleLoanUpdate(LoanDTO loan) {
     }
   }
 
+  @Override
+  public void handleRequestUpdate(RequestDTO requestDTO) {
+    var transaction = transactionRepository.fetchOneByRequestId(requestDTO.getId()).orElse(null);
+    if (transaction == null) {
+      return;
+    }
+
+    TransactionHold transactionItemHold = transaction.getHold();
+    if (!transactionItemHold.getFolioItemId().equals(requestDTO.getItemId())) {
+      log.info("Updating transaction {} on request to transfer {}", transaction.getId(),requestDTO.getId());
+      transactionItemHold.setFolioItemId(requestDTO.getItemId());
+      transaction.setState(TRANSFER);
+      reportTransferRequest(transaction, requestDTO.getItemId().toString());
```

Comment:
we link items by hrid when contributing them to central server, so I assume here it also should be referenced by hrid. Ask PO about this. 
Also, check the rules for FOLIO to D2IR fields mapping https://wiki.folio.org/display/FOLIJET/SPIKE%3A+%5BMODINREACH-79%5D+Record+Contribution%3A+Analyze+data+required+for+D2IR+contribution

Reply:
Fixed